packages/client/hmi-client/src/services/code.ts

```diff
@@ -4,6 +4,39 @@ import { Ref } from 'vue';
 import { addAsset } from '@/services/project';
 import { logger } from '@/utils/logger';
 
+async function getCodeAsset(codeAssetId: string): Promise<Code | null> {
+	const response = await API.get(`/code-asset/${codeAssetId}`);
+
+	if (!response || response.status >= 400) {
+		logger.error('Error getting code file as text');
+		return null;
+	}
+
+	return response.data as Code;
+}
+
+async function updateCodeAsset(
+	code: Code,
+	file: File,
+	progress: Ref<number>
```

Comment:
What is this `progress`?

Reply:
It's required in `addFileToCodeAsset` which is being called in this method

Reply:
Then why are we not displaying it in the button?
If this would take too long, could you write an issue to add this feature please?